File: src/main/java/vn/edu/hust/vha/hims/modules/contract/controller/ContractController.java
package vn.edu.hust.vha.hims.modules.contract.controller;

import java.util.UUID;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import vn.edu.hust.vha.hims.modules.contract.dto.ContractCreateDTO;
import vn.edu.hust.vha.hims.modules.contract.dto.ContractResponseDTO;
import vn.edu.hust.vha.hims.modules.contract.service.ContractService;

@RestController
@RequestMapping("/api/v1/contracts")
@RequiredArgsConstructor
public class ContractController {

    private final ContractService contractService;

    /**
     * Tạo hợp đồng mới
     * POST /api/v1/contracts
     */
    @PostMapping
    public ResponseEntity<ContractResponseDTO> createContract(
            @Valid @RequestBody ContractCreateDTO request) {
        
        ContractResponseDTO response = contractService.createContract(request);
        return ResponseEntity.status(HttpStatus.CREATED).body(response);
    }

    /**
     * Lấy thông tin hợp đồng theo ID
     * GET /api/v1/contracts/{id}
     */
    @GetMapping("/{id}")
    public ResponseEntity<ContractResponseDTO> getContractById(
            @PathVariable UUID id) {
        
        ContractResponseDTO response = contractService.getContractById(id);
        return ResponseEntity.ok(response);
    }
}

File: src/controllers/contract.controller.js
import * as contractService from '../services/contract.service.js';

export const getAllContracts = async (req, res, next) => {
    try {
        const contracts = await contractService.getAllContracts();
        res.json(contracts);
    } catch (error) {
        next(error);
    }
};

export const getContractById = async (req, res, next) => {
    try {
        const contract = await contractService.getContractById(req.params.id);
        if (!contract) {
            return res.status(404).json({ error: 'Contract not found' });
        }
        res.json(contract);
    } catch (error) {
        next(error);
    }
};

export const createContract = async (req, res, next) => {
    try {
        const contract = await contractService.createContract(req.body);
        res.status(201).json(contract);
    } catch (error) {
        next(error);
    }
};

export const updateContract = async (req, res, next) => {
    try {
        const contract = await contractService.updateContract(req.params.id, req.body);
        if (!contract) {
            return res.status(404).json({ error: 'Contract not found' });
        }
        res.json(contract);
    } catch (error) {
        next(error);
    }
};
